### DISCUSSION TITLE
Docker Images

### DISCUSSION THREAD

**Discussion by foxhoundv** [ltsp/ltsp#386] [Ideas]
Hey @alkisg, it's been a while since we last spoke. How are things going? Did you get the funding needed yet? Anyways, I had an idea that I thought I would pass on, it may make things easier... maybe.

What if you had a docker image for LTSP that could allow incorporation of the images to push? I don't know how familiar you are with Docker, but a Container can be connected to a persistent volume. If I'm not mistaken, the images that LTSP can load can be stored in the persistent volume.

**Comment by alkisg** [comment]
Hi @foxhoundv, I located an IRC discussion with you a year ago, I think you're asking about Raspberry Pi OS support? I did add LTSP support for that but without any funding.

Regarding docker, I didn't really understand how it would be useful. People can manage their images however they want, with VirtualBox, or with Vagrant, or by fetching them from osboxes.org... And an `ltsp image xxx` command will make them available to the clients.

Are you looking to replace the simple `ltsp image xxx` command with a complex docker infrastructure?

**Reply by foxhoundv** [reply]
@alkisg Sorry for the late response, I somehow missed the email telling me that you responded. Anyways, what I was meaning was that I remember that I had to have you remote into my server to help finalize the setup to allow the LTSP image Raspberry Pi to work. In my understanding of Docker, the image is built with layers that basically are kind of like a setup script. Would it be possible to make an LTSP image for Docker that would already be set with the basics, and in the environment, the docker user could specify network parameters that would then get passed into the container upon deployment? I know I am horrible when it comes to giving the proper names of things. I hope I am saying all this in a way that you are able to understand what I am trying to convey.

**Comment by alkisg** [comment]
@foxhoundv, I see no benefit in using docker to manually do things that are done by LTSP automatically. It makes things more difficult, not easier. For setting up networking information in dnsmasq, nfs etc, the `ltsp dnsmasq` and `ltsp nfs` commands can be used, while docker doesn't have such tools.
Personally I have no plans to work on anything related to docker.

The "manual ltsp setup" a year ago, was because the Raspberry Pi support in LTSP wasn't finalized. Docker wouldn't help there at all. The solution was to add code in LTSP itself.

If one wants to implement an `ltsp pnp` command that autodetects everything, that is doable.
But in my personal roadmap, the next big thing is to add web services in LTSP.

**Reply by foxhoundv** [reply]
@alkisg Ok, thank you for the information. It was just a curiosity.

I am excited to see what the web services in LTSP has to offer. Do you mean like a web interface for controlling the LTSP environment?

**Comment by alkisg** [comment]
Check the blueprint there: https://github.com/ltsp/ltsp/issues/147

**Reply by foxhoundv** [reply]
Wow, a good chunk of that is above my head.
